lib/views/setting/setting_view.dart

```diff
@@ -126,63 +116,140 @@ class _SettingViewState extends State<SettingView> {
-                  isLoading = true;
+                  setting.nickName = name;
                 });
-                saveSettings().then((value) => {
-                      isLoading = false,
-                    });
-                widget.provider.registerLazyDependence<AppApi>(() {
-                  if (setting.connectionType == clients[0]) {
-                    return CLNApi(
-                        mode: ClientMode.grpc,
-                        client: ClientProvider.getClient(
-                            mode: ClientMode.grpc,
-                            opts: {
-                              ///FIXME: make a login page and take some path as input
-                              'certificatePath': setting.path,
-                              'host': setting.host,
-                              'port': 8001,
-                            }));
-                  } else {
+              },
+              initialValue: setting.nickName == "null" ? "" : setting.nickName,
+              decoration: const InputDecoration(
+                border: OutlineInputBorder(),
+                hintText: 'My lightning node',
+              ),
+            ),
+            _buildCorrectSettingView(context: context, setting: setting),
+            ElevatedButton(
+              onPressed: () {
+                if (setting.isValid()) {
+                  saveSettings();
+                  widget.provider.registerLazyDependence<AppApi>(() {
                     return CLNApi(
-                        mode: ClientMode.unixSocket,
-
-                        ///FIXME: make a login page and take some path as input
+                        mode: setting.clientMode,
                         client: ClientProvider.getClient(
-                            mode: ClientMode.unixSocket,
-                            opts: {
-                              // include the path if you want use the unix socket. N.B it is broken!
-                              'path': "${setting.path}/lightning-rpc",
-                            }));
-                  }
-                });
-                Navigator.of(context).pushAndRemoveUntil(
-                    MaterialPageRoute(
-                        builder: (context) => HomeView(
-                              provider: widget.provider,
-                            )),
-                    (Route<dynamic> route) => false);
-              }
-            },
-            child: const Text("Save"),
-          ),
-        ],
+                            mode: setting.clientMode, opts: setting.toOpts()));
+                  });
+                  Navigator.of(context).pushAndRemoveUntil(
+                      MaterialPageRoute(
+                          builder: (context) => HomeView(
+                                provider: widget.provider,
+                              )),
+                      (Route<dynamic> route) => false);
+                }
+              },
+              child: const Text("Save"),
```

Comment:
Now that you have the setting fixed, you could also add a new button to clean see setting? with `await preferences.remove('setting');`

Reply:
ooh yes will add this!